cobbler/modules/managers/isc.py

```diff
@@ -56,23 +54,17 @@ def __init__(self, api):
         self.settings_file_v4 = utils.dhcpconf_location(utils.DHCP.V4)
         self.settings_file_v6 = utils.dhcpconf_location(utils.DHCP.V6)
 
-    def write_v4_config(self):
+    def write_v4_config(self, template_file="/etc/cobbler/dhcp.template"):
         """
         DHCPv4 files are written when ``manage_dhcp_v4`` is set in our settings.
+
+        :param template_file: The location of the DHCP template.
         """
 
-        template_file = "/etc/cobbler/dhcp.template"
         blender_cache = {}
 
-        # TODO: Has to be refactored.
-        try:
-            f2 = open(template_file, "r")
-        except Exception:
-            raise CX("error reading template: %s" % template_file)
-
-        template_data = ""
-        template_data = f2.read()
-        f2.close()
+        with open(template_file, "r") as f2:
```

Comment:
Are we not longer taking in care of the exceptions that reading this file could produce?

Reply:
Well my opinion on this is the following:

We do care about the exception. Just not catching it. In case we cannot read files under our own control we should just loudly fail. Of course we should cleanup but in this case the cleanup is just closing the file. So no I think in this case the context manager is a better idea instead of the manual `.close()`.

P.S.: We have a ticket open which will add checks on daemon startup if all required files are readable and/or writable. Additionally I plan on making those checks explicitly available via `cobbler check`. So although not explicitly handled on every case I plan on adding global checks soon. The reason for this strategy is that I don't want to add error handling to all 70 files we regularly manage inside the code.

Reply:
Ack. Thanks for claryfing this :+1: